### PR TITLE
Presto on Spark log functions invoked in query

### PR DESCRIPTION
**Pull request by JasmineWeii**
## Change
QueryInfo.scalarFunctions, aggregateFunctions and windowsFunctions is empty for Presto on Spark jobs.
This PR grab the functions invoked from the initial query sql text analysis result and propagate it to QueryInfo object.

## Test plan
Existing unit tests. QueryInfo.scalarFunctions, aggregateFunctions and windowsFunctions are being populated properly.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Comment by Ali-P** [comment]
Thanks @JasmineWeii ! LGTM.
QQ: Do you intend to capture external functions (e.g. JSON-registered) as well? If so, does this solution capture them as well?

**Comment by JasmineWeii** [comment]
> QQ: Do you intend to capture external functions (e.g. JSON-registered) as well? If so, does this solution capture them as well?

I am interpreting your question as whether user defined functions are captured or not. From what I learnt from @zacw7, is that functions invoked from Analyzer is not differentiated by how it's being registered but the function type only. And there are only 3 function type for Presto.

**Comment by vanekjar** [comment]
I noticed that for a regular Presto query, this feature is guarded by a system property. Should we do the same here to make sure it's aligned?

https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/execution/SqlQueryExecution.java#L230

File: presto-spark-base/src/main/java/com/facebook/presto/spark/planner/PrestoSparkQueryPlanner.java
Comment (vanekjar) [inline presto-spark-base/src/main/java/com/facebook/presto/spark/planner/PrestoSparkQueryPlanner.java:153]: Is there a need to create the empty `HashSet` instance? Maybe `Collections.emptySet()` would be sufficient.

File: presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java
Comment (shrinidhijoshi) [inline presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java:375]: Let's add the `isLogInvokedFunctionNamesEnabled(getSession())` check to maintain consistent behavior with Presto

File: presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java
Comment (zacw7) [inline presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java:125]: Unused import.

File: presto-spark-base/src/main/java/com/facebook/presto/spark/planner/PrestoSparkQueryPlanner.java
Comment (zacw7) [inline presto-spark-base/src/main/java/com/facebook/presto/spark/planner/PrestoSparkQueryPlanner.java:151]: Let's do static import `SCALAR` and remove the prefix:
```suggestion
Set<String> invokedScalarFunctions = functionsInvoked.getOrDefault(SCALAR, new HashSet<>());
```

Same for `AGGREGATE` and `WINDOW`.

**Review by zacw7 — COMMENTED** [review]
LGTM % nits.

Consider adding some tests if possible. Something like: https://github.com/prestodb/presto/blob/00ed8e236af5598da2e856d2d1e35f234a053a9d/presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java#L5906-L5942

File: presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java
Comment (JasmineWeii) [reply presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java:375]: I am adding this condition checks when PlanAndMore object is constructed, since the Presto code conditionally fetch those function information in stateMachine instead of QueryExecution.

**Review by zacw7 — APPROVED** [review]

**Review by Ali-P — APPROVED** [review]
LGTM! I suggest waiting a bit to see if others have other feedback! :)

**Review by Ali-P — APPROVED** [review]

**Review by vanekjar — APPROVED** [review]
LGTM, thanks for this improvement!

**Comment by rschlussel** [comment]
hey, this change did not have committer/codeowner approval yet. How did it get merged? It shouldn't be possible.

File: presto-spark-base/src/main/java/com/facebook/presto/spark/planner/PrestoSparkQueryPlanner.java
Comment (rschlussel) [inline presto-spark-base/src/main/java/com/facebook/presto/spark/planner/PrestoSparkQueryPlanner.java:158]: use `ImmutableSet.of()` in general rather than Collections.emptySet()

**Pull request merged**
